packages/patrol_cli/lib/src/features/drive/drive_command.dart

```diff
@@ -248,34 +228,56 @@ class DriveCommand extends StagedCommand<DriveCommandConfig> {
     );
 
     var exitCode = 0;
-    for (final target in config.targets) {
-      _testRunner.addTest((device) async {
-        if (_disposeScope.disposed) {
-          _logger.fine('Skipping running $target...');
-          return;
+    _testRunner
+      ..builder = (target, device) async {
+        try {
+          await _flutterTool.build(target, device);
+        } catch (err) {
+          exitCode = 1;
```

Comment:
For now this crude way of handling errors is OK.

Refactoring this is in #561's scope.